build/spas.ts

```diff
@@ -304,7 +305,7 @@ export async function buildSPAs(options: {
                   tag: parents.length > 2 ? parents[1] : null,
                 };
               }
-            } else if (segment === "blog") {
+            } else if (segment === "blog" && BLOG_ROOT) {
```

Comment:
Not needed anymore since we keep returning `null`
```suggestion
            } else if (segment === "blog") {
```